test/distributed/_shard/sharding_plan/test_sharding_plan.py

```diff
@@ -0,0 +1,149 @@
+
+# Owner(s): ["oncall: distributed"]
+import sys
+
+import torch
+import torch.nn as nn
+from torch.testing._internal.common_distributed import (
+    requires_nccl,
+    skip_if_lt_x_gpu,
+)
+from torch.distributed._shard import shard_module, ShardedModuleSwapper
+from torch.distributed._shard.sharding_plan import ShardingPlan, ShardingPlanner
+from torch.distributed._shard.sharding_spec import ChunkShardingSpec
+from torch.distributed._shard.sharded_tensor import ShardedTensor
+
+from torch.testing._internal.common_utils import TEST_WITH_DEV_DBG_ASAN
+from torch.testing._internal.distributed._shard.sharded_tensor import (
+    TEST_GPU_NUM,
+    ShardedTensorTestBase,
+    with_comms,
+)
+from torch.testing._internal.distributed._shard.sharded_tensor._test_ops_common import (
+    generate_chunk_sharding_specs_for_test,
+)
+
+if TEST_WITH_DEV_DBG_ASAN:
+    print(
+        "Skip dev-asan as torch + multiprocessing spawn have known issues",
+        file=sys.stderr,
+    )
+    sys.exit(0)
+
+
+class SimpleMegatronLM(nn.Module):
+    def __init__(self, linear_size, rank=None):
+        super().__init__()
+        self.fc1 = nn.Linear(*linear_size[0])
+        self.gelu = nn.GELU()
+        self.fc2 = nn.Linear(*linear_size[1])
+        if rank:
+            self.fc1.cuda(rank)
+            self.fc2.cuda(rank)
+
+    def forward(self, inp):
+        return self.fc2(self.gelu(self.fc1(inp)))
```

Comment:
Maybe move this to util or `test_common`? My guess is that we might need this very often down the road.